internal/controller/ocivalidator_controller.go

```diff
@@ -127,15 +127,28 @@ func (r *OciValidatorReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-// secretKeyAuth retrieves the username and password from the secret referenced in the rule's auth field.
+// auth retrieves the username and password from the rule's auth field.
+// If a secretName is provided in the rule's auth field, then the secret is fetched and the username and password are retrieved from the secret.
 // Any additional key-value pairs in the secret are set as environment variables, to be picked up by auth keychains (e.g. ECR, Azure).
-func (r *OciValidatorReconciler) secretKeyAuth(req ctrl.Request, rule v1alpha1.OciRegistryRule) (string, string, error) {
-	if rule.Auth.SecretName == "" {
-		return "", "", nil
+func (r *OciValidatorReconciler) auth(req ctrl.Request, rule v1alpha1.OciRegistryRule) (string, string, error) {
+	if rule.Auth.SecretName != nil {
```

Comment:
What if `Auth.SecretName` is a non-nil pointer to an empty string?

Reply:
adding back the original behaviour of returning "", "", nil